adapters/consumable/consumable.go

```diff
@@ -139,31 +145,45 @@ func (a *ConsumableAdapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *
 	gdpr := bidGdpr{}
 
 	ccpaPolicy, err := ccpa.ReadFromRequest(request)
-	if err == nil {
-		body.CCPA = ccpaPolicy.Consent
+	if err != nil {
+		errs = append(errs, err)
 	}
 
+	body.CCPA = ccpaPolicy.Consent
+
 	// TODO: Replace with gdpr.ReadPolicy when it is available
 	if request.Regs != nil && request.Regs.Ext != nil {
 		var extRegs openrtb_ext.ExtRegs
-		if err := json.Unmarshal(request.Regs.Ext, &extRegs); err == nil {
-			if extRegs.GDPR != nil {
-				applies := *extRegs.GDPR != 0
-				gdpr.Applies = &applies
-				body.GDPR = &gdpr
-			}
+		if err := json.Unmarshal(request.Regs.Ext, &extRegs); err != nil {
+			errs = append(errs, err)
+		}
+		if extRegs.GDPR != nil {
+			applies := *extRegs.GDPR != 0
+			gdpr.Applies = &applies
+			body.GDPR = &gdpr
 		}
 	}
 
 	// TODO: Replace with gdpr.ReadPolicy when it is available
 	if request.User != nil && request.User.Ext != nil {
 		var extUser openrtb_ext.ExtUser
-		if err := json.Unmarshal(request.User.Ext, &extUser); err == nil {
-			gdpr.Consent = extUser.Consent
-			body.GDPR = &gdpr
+		if err := json.Unmarshal(request.User.Ext, &extUser); err != nil {
+			errs = append(errs, err)
 		}
+		gdpr.Consent = extUser.Consent
+		body.GDPR = &gdpr
 	}
 
+	if request.Source != nil && request.Source.Ext != nil {
+		var extSChain openrtb_ext.ExtRequestPrebidSChain
+		if err := json.Unmarshal(request.Source.Ext, &extSChain); err != nil {
+			errs = append(errs, err)
+		}
+		body.SChain = extSChain.SChain
+	}
+
+	body.Coppa = request.Regs != nil && request.Regs.COPPA > 0
```

Comment:
I'm confused as to what this line of code is trying to accomplish, like what value is `body.Coppa` being set to?